### ISSUE TITLE
[QST] Is it possible to use early stopping in Transformers4rec?

### ISSUE THREAD

**Issue by antklen** [NVIDIA-Merlin/Transformers4Rec#516]
I'm trying to use EarlyStoppingCallback from huggingface transformers with code like this:
```python
from transformers import EarlyStoppingCallback
from transformers4rec.config.trainer import T4RecTrainingArguments
from transformers4rec.torch import Trainer

training_args = T4RecTrainingArguments(
            data_loader_engine='nvtabular',
            output_dir="./tmp",
            max_sequence_length=64,
            num_train_epochs=10,
            save_steps=100000,
            evaluation_strategy='steps',
            eval_steps=1000,
            load_best_model_at_end=True,
            metric_for_best_model='/next-item/avg_precision_at_100',
            greater_is_better=True,
        )

early_stopping = EarlyStoppingCallback(early_stopping_patience=1)

trainer = Trainer(
    model=model,
    args=training_args,
    schema=schema,
    train_dataset_or_path='train.parquet',
    eval_dataset_or_path='valid.parquet',
    compute_metrics=True,
    callbacks=[early_stopping]
)

trainer.train()
```
but when metric value becomes lower nothing happened. I found that after training `trainer.state.best_metric` is None. May be this  is the problem because it is checked here https://github.com/huggingface/transformers/blob/v4.24.0/src/transformers/trainer_callback.py#L549.

So is there any way to make early stopping?

**Comment by rnyak** [comment]
@antklen you might want to check this examples and try to integrate it: https://wandb.ai/ayush-thakur/huggingface/reports/Examples-of-Early-Stopping-in-HuggingFace-Transformers--Vmlldzo0MzE2MTM
